src/navigation/breadcrumbs/breadcrumbs.jsx

```diff
@@ -0,0 +1,203 @@
+                    transition: 'color 0.1s',
+                    '&:hover': {
+                        '& .icon-use-path': {
+                            fill: colorHighlight,
+                        },
+                    },
+                    '&_is_last': {
+                        '& div': {
+                            fontWeight: fontWeightRegular,
+                        },
+                        [`& .${BEM_NAVIGATION_BREADCRUMBS}--breadcrumb_separator`]: {
+                            padding: '0 5px 0 0',
+                        },
+                    },
+                    [`& .${BEM_NAVIGATION_BREADCRUMBS}--breadcrumb_typography`]: {
+                        display: 'inline',
+                        '& > div': {
+                            display: 'inline',
+                        },
+                    },
+                },
+                '&:hover': {
+                    [`& .${BEM_NAVIGATION_BREADCRUMBS}--breadcrumb_is_last`]: {
+                        color: textColorSecondary,
+                        '&:hover': {
+                            color: textColorPrimary,
+                        },
+                    },
+                },
+            },
+        },
+    };
+});
+
+function Breadcrumbs(props) {
+    const {
+        router,
+        separatorIconType,
+        showOnlyPreviousRoute,
+    } = props;
+
+    const classes = useStyles(props);
+    const [existentRoutes, setExistentRoutes] = useState([]);
+    const [pathBreadcrumbs, setPathBreadcrumbs] = useState([]);
+    const pathName = get(router, 'location.pathname');
+    const pathParams = get(router, 'params');
+    const prevPathName = useRef();
+    const routerPushFunction = get(router, 'push');
+
+    useEffect(() => {
+        const routes = get(router, 'routes', []);
+        const currentExistentRoutes = routesToArray(routes);
+        setExistentRoutes(currentExistentRoutes);
+        const currentBreadCrumbs = getPathNameBreadcrumbs(
+            pathName,
+            pathParams,
+            routerPushFunction,
+            currentExistentRoutes,
+        );
+        setPathBreadcrumbs(currentBreadCrumbs);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []); // didMount
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    useEffect(() => {
+        const shouldUpdatePathBreadcrumbs = prevPathName.current &&
+            prevPathName.current !== pathName;
+
+        if (shouldUpdatePathBreadcrumbs) {
+            const currentBreadCrumbs = getPathNameBreadcrumbs(
+                pathName,
+                pathParams,
+                routerPushFunction,
+                existentRoutes,
+            );
+
+            setPathBreadcrumbs(currentBreadCrumbs);
+        }
+
+        prevPathName.current = pathName;
+    }); // didUpdate
+
+    return (
+        <div className={classes.root}>
+            <ul className={`${BEM_NAVIGATION_BREADCRUMBS}--list`}>
```

Comment:
Keep the className, but please use classes for the JSS instead.

Reply:
done

Reply:
I am not sure I am seeing the change here.